### PR TITLE
DSD-1458: Styleguide docs to storybook7

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1458](https://jira.nypl.org/browse/DSD-1458)

## This PR does the following:

- Updates all Styleguide docs to Storybook 7 version.
- Please take a close look at all the doc pages especially the "Colors" one which had additional bug fixes.
- An example production page: https://nypl.github.io/nypl-design-system/reservoir/v1/?path=/story/style-guide-bidirectionality--page

Bug fixes:
- There was a color name typo in the theme object which did not allow the color to render properly.

Screen Shot 2023-05-10 at 4 25 46 PM

- Some props were not named correctly and _maybe_ did not render props correctly. @bigfishdesign13 let me know if the additional values in the table are correct. Otherwise, the data should be updated and removed. Specifically, this is for the `ui.typography.heading` and `ui.typography.body` in `dark.ui.bg.page`, `dark.ui.bg.default`, `dark.ui.bg.hover`, and `dark.ui.bg.active`.

Screen Shot 2023-05-10 at 4 26 30 PM

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -53,7 +53,14 @@ const config: StorybookConfig = {
     "../src/components/HorizontalRule/HorizontalRule.mdx",
     "../src/components/StatusBadge/StatusBadge.stories.tsx",
     "../src/components/StatusBadge/StatusBadge.mdx",
-    "../src/docs/Welcome.mdx",
```
Comment (bigfishdesign13) [inline .storybook/main.ts]: Did you intend to remove this?

File: src/components/StyleGuide/Bidirectionality.mdx [hunk 2]
```diff
@@ -163,7 +166,7 @@ directionality should be implemented at the application level. The following are
 general guidelines on how to set up directionality but implementation will be up
 to the application and its architecture.
 
-### `dir` HTML attribute
+### \`dir\` HTML attribute
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/Bidirectionality.mdx]: Ha!  I ran into a problem with this formatting in another story.  I took the easy way out and just removed the backticks.

File: src/components/StyleGuide/ColorCard.tsx [hunk 3]
```diff
@@ -415,14 +1215,20 @@ export const ColorCard = (props: React.PropsWithChildren<ColorCardProps>) => {
...
+          >
             {backgroundColor}
           </Heading>
-          <Text fontWeight="medium" noSpace size="tag">
+          <Text fontWeight="medium" noSpace size="tag" m="0" color={textColor}>
             {"CSS: var(" + cssVarName + ")"}
           </Text>
           {colorSource && (
-            <Text fontWeight="medium" noSpace size="tag">
+            <Text fontWeight="medium" noSpace size="tag" color={textColor}>
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/ColorCard.tsx]: Despite the `noSpace` prop, the top margin is still being applied.

File: src/components/StyleGuide/ColorCard.tsx [hunk 4]
```diff
@@ -204,11 +1002,13 @@ export const DataTable = (props: React.PropsWithChildren<DataTableProps>) => {
       td: {
         fontSize: "text.tag",
         ...cellStyles,
+        borderTop: "1px solid !important",
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/ColorCard.tsx]: Ideally, the border should be applied to the bottom of the elements, as there should be a bottom border for the last row.  But that would mean you'll probably need to also add some styles for thead to ensure that the header row also has a bottom border.

File: src/components/StyleGuide/Forms.mdx [hunk 5]
```diff
@@ -1,11 +1,9 @@
...
 
 import Card from "../Card/Card";
 import List from "../List/List";
-import { getCategory } from "../../utils/componentCategories";
-import DSProvider from "../../theme/provider";
 
-<Meta title={getCategory("Forms")} />
+<Meta title="Style Guide/Forms" />
 
 # Forms
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/Forms.mdx:8]: In the `Form Input Components` list, can you change `Dropdown` to `MultiSelect` and then re-alphabetize?

File: src/components/StyleGuide/Iconography.mdx [hunk 6]
```diff
@@ -31,16 +29,20 @@ The graphic of the `Icon` component is handled through the `name` prop. To see
...
+>
+  <Stack spacing="s" direction="row">
+    <Icon decorative size="xxxlarge" name="check" />
+    <Icon decorative size="xxxlarge" name="close" />
+    <Icon decorative size="xxxlarge" name="errorOutline" />
+    <Icon decorative size="xxxlarge" name="speakerNotes" />
+  </Stack>
+</Box>
 
 | Values for Examples Above |
```
Comment (bigfishdesign13) [inline src/components/StyleGuide/Iconography.mdx:47]: I just realized that the nice table formatting is gone.  Do you know we can get that back?  Do we need to add our own styles?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some minor requests and some questions.

**Comment by EdwinGuzman** [comment]
I need to make another round of updates because I worked on getting the table to look right in the `ColorCard` table that it messes with table styles on all other storybook doc pages.

**Comment by EdwinGuzman** [comment]
Latest update should cover all feedback updates.

**Review by bigfishdesign13 — APPROVED** [review]
Looks great.

**Pull request merged**